### DISCUSSION TITLE
Generate different Best Models everytime when apply compare_models on same dataset

### DISCUSSION THREAD

**Discussion by akashg116414** [pycaret/pycaret#1250] [Q&A] [answered]
Hello,

if the data_split_shuffle = True then compare_models giving different best model everytime on same dataset and if data_split_shuffle = False then it is giving same best model but fold_strategy = "Kfold" is not working as it is giving NAN value in Kappa column and all other value like precision, f-score, AUC and Accuracy are coming same for all model and some time only 2-3 models are generating in case of fold_strategy = "Kfold" and data_split_shuffle = False.

Please find the attachment of the notebook for this problem.
[pycaret_test_train_split.zip](https://github.com/pycaret/pycaret/files/6470475/pycaret_test_train_split.zip)

**Comment by ngupta23** [comment, marked as answer]
@akashg116414 Can you set the session id to a fixed number in the setup? This acts as a seed for the pipeline. If you do not specify it, a different number is selected in each run leading to randomness in the output.

From [this](https://pycaret.org/classification/)

> session_id: int, default = None
If None, a random seed is generated and returned in the Information grid. The unique number is then distributed as a seed in all functions used during the experiment. This can be used for later reproducibility of the entire experiment.

**Comment by pycaret** [comment]
@akashg116414 Seems like you are not using the `session_id` parameter in the `setup` function. That is why every time you run `setup`  you get a different `train_test_split`. 

Hope this helps.

**Comment by akashg116414** [comment]
And what about NAN value in Kappa column and all other value like precision, f-score, AUC and Accuracy are coming same for all model and some time only 2-3 models are generating in case of fold_strategy = "Kfold" and data_split_shuffle = False.

**Reply by pycaret** [reply]
@akashg116414 That will be the problem with Kappa metric scorer from sklearn.
